scala-generator/src/main/scala/models/KafkaConsumer.scala

```diff
@@ -163,49 +163,63 @@ package ${ssd.namespaces.base}.kafka {
     def processBatchThenCommit(
       processor: Map[String, Seq[${className}]] ⇒ scala.util.Try[Map[String, Seq[${className}]]],
       batchSize: Int = 1
-    ): scala.util.Try[Map[String, Seq[${className}]]] = {
+    ): scala.util.Try[Map[String, Seq[${className}]]] =
+      doProcess[${className}] { message ⇒
+        Option(message.message).map(Json.parse(_).as[${className}])
+      }(processor, batchSize)
+
+    def processBatchWithKeysThenCommit(
+      processor: Map[String, Seq[(String, Option[${className}])]] ⇒ scala.util.Try[Map[String, Seq[(String, Option[${className}])]]],
+      batchSize: Int = 1
+    ): scala.util.Try[Map[String, Seq[(String, Option[${className}])]]] =
+      doProcess[(String,  Option[${className}])] { message ⇒
+        Some(
+          message.key → Option(message.message).map(Json.parse(_).as[${className}])
+        )
+      }(processor, batchSize)
+
+    def doProcess[T](
```

Comment:
It needs to be as it's used by both `processBatchThenCommit` and `processBatchWithKeysThenCommit` and they have different types of messages.